### DISCUSSION TITLE
PLSR with spectral and experimented data in Matlab.

### DISCUSSION THREAD

**Discussion by NWK-programme** [community/community#131427] [Programming Help]
### Body

Hi!
I am trying to perform PLSR on a large amount of spectral data: 300000+ data points, 112 wavebands and 4 classes. From these samples, I did a moisture analysis of which I got n=72 (36 samples in duplicate). I thought it was possible to do PLSR on this set, but I get the error stating: "rows of X and Y should match". Is there a way to fix this? Or is it not possible to do PLSR on this set? I work in MATLAB and got my data from the spectral program Perclass Mira.

Thanks in advance!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by uttambodara** [comment]
Response #1

**Comment by Dawoodoz** [comment]
First step is to calculate the expected result manually for a small subset of the data, before getting biased from what you see returned from the computer. Read any documentation of the library used, to make sure that the data is supplied in the correct format. Then let it run the same calculation that you did manually, to assert that you understood the documentation correctly. If it works, then you can provide your whole data and plot the approximation over a graph for confirmation. If it does not work the way you wanted, read the code to figure out the different use case and write your own formula by combining basic operations.

Sometimes you may need to provide a linearly increasing vector for the time to make one dimensional data two dimensional, so that a curve can be approximated, but I do not know the format from Mira nor what you are trying to measure.

**Reply by NWK-programme** [reply]
The data format provided by Mira is three dimensional: (amount of measured pixels, +- 300000) x (amoount of wavebands) x (sample groups, in my case 4). But after thinking about it, I might need to make an average spectrum value of the 300000+ pixels x the specific waveband range to get the same amount of rows for both my X (responses) and Y (predictors). I am not sure if this is the correct way, but this is what I was thinking about. I have already calculated my expected results and made a average of every duplicate.

I am curious what you think about it.

**Reply by Dawoodoz** [reply]
It completely depends on what kind of patterns you are trying to find in the data, but a common approach when you have no clue is to split data into classes based on criterias to reduce the number of dimensions and then plot it with curve fitting along remaining attributes until the data makes sense. Just don't select criterias so specific that you overfit, so it is good practice to have another set of data to confirm any patterns you saw.
